src/python/gudhi/simplex_tree.pyx

```diff
@@ -90,24 +90,15 @@ cdef class SimplexTree:
             (with more :meth:`assign_filtration` or
             :meth:`make_filtration_non_decreasing` for instance) before calling
             any function that relies on the filtration property, like
-            :meth:`initialize_filtration`.
+            :meth:`persistence`.
         """
         self.get_ptr().assign_simplex_filtration(simplex, filtration)
 
     def initialize_filtration(self):
         """This function initializes and sorts the simplicial complex
         filtration vector.
 
-        .. note::
-
-            This function must be launched before
-            :func:`persistence()<gudhi.SimplexTree.persistence>`,
-            :func:`betti_numbers()<gudhi.SimplexTree.betti_numbers>`,
-            :func:`persistent_betti_numbers()<gudhi.SimplexTree.persistent_betti_numbers>`,
-            or :func:`get_filtration()<gudhi.SimplexTree.get_filtration>`
-            after :func:`inserting<gudhi.SimplexTree.insert>` or
-            :func:`removing<gudhi.SimplexTree.remove_maximal_simplex>`
-            simplices.
+        .. deprecated:: 3.2.0
         """
         self.get_ptr().initialize_filtration()
```

Comment:
If deprecated, shouldn't we display a message ?

Reply:
We could. One reason to warn would be because we may remove it in a later release, but I don't really have plans for that. On the other hand, for downstream users who try to be compatible with several releases of gudhi, these warnings can be a bit irritating. Maybe we can have several steps: one or more releases where the function is unnecessary, then one or more releases where it warns, and finally a release where it is removed?
```python
if getattr(gudhi, "_warn_deprecated_ST_initialize_filtration", True):
    print("SimplexTree.initialize_filtration() is now unnecessary and deprecated", file=sys.stderr)
    gudhi._warn_deprecated_ST_initialize_filtration = False
```